core/src/main/java/org/apache/iceberg/SystemConfigs.java

```diff
@@ -43,14 +43,14 @@ private SystemConfigs() {}
           Integer::parseUnsignedInt);
 
   /**
-   * Sets the size of the delete worker pool. This limits the number of threads used to compute the
-   * PositionDeleteIndex from the position deletes for a data file.
+   * Sets the size of the delete worker pool. This limits the number of threads used to read delete
+   * files for a data file.
    */
   public static final ConfigEntry<Integer> DELETE_WORKER_THREAD_POOL_SIZE =
       new ConfigEntry<>(
           "iceberg.worker.delete-num-threads",
           "ICEBERG_WORKER_DELETE_NUM_THREADS",
-          Math.max(2, Runtime.getRuntime().availableProcessors()),
+          Math.max(2, 4 * Runtime.getRuntime().availableProcessors()),
```

Comment:
Not a huge deal but we are avoiding the RevCheck here by putting our multiplier in a constant here. We should probably move the 4 into a field so future modifications trigger the Rev checker

Reply:
I can do that but is it something we want to expose to others? Would the goal be to bring the attention or to prohibit future modifications?

Reply:
You can keep it private, it's to prohibit future mods

Reply:
Looks like we don't do that for other properties as well.
I'd be open to explore that but for all properties in a separate PR.